StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/build/MSBuild15.props

```diff
@@ -0,0 +1,35 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<!--
+  Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+  Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Condition="Exists('stylecop.json')">
+    <!--
+      If stylecop.json exists in the project directory add it to <AdditionalFiles /> automatically.  The point of
+      this is that the user sees the file in Visual Studio Solution Explorer but they don't understand that they
+      need to add it to the <AdditionalFiles /> item group.
+      
+      1. Legacy project system - Files in the project directory are in an explicit <None /> item.
+      2. New project system - Files in the project directory are implicitly added to the <None /> item through a glob.
+    -->
+    <None Remove="stylecop.json" />
+
+    <AdditionalFiles Include="stylecop.json" Condition=" '@(AdditionalFiles->EndsWith('stylecop.json', System.StringComparison.OrdinalIgnoreCase))' != 'true' " />
+  </ItemGroup>
+
+  <ItemGroup Condition="!Exists('stylecop.json')">
```

Comment:
❓ Why separate this into the two cases? It seems like it would be simpler to combine the two cases into one.

Reply:
If `stylecop.json` is in the project directory, there is almost guaranteed to already be a `<None />` item pointing to it.  I wanted to treat this case slightly differently by moving it to `<AdditionalItems />`.  However, if there is a `stylecop.json` in a parent folder, it gets added with a `Link` and any existing items are left alone.  

Technically I could combine them, I was just trying to be more exact.  I'm worried about the upgrade experience for users as they try out new versions of the package.  They already would have done the manual steps and upgrading could cause duplicates to show up in Solution Explorer.  If you're fine with some release notes documentation or something, I can simplify some of this.

Reply:
> If you're fine with some release notes documentation or something, I can simplify some of this.

If we get all the edge cases working, I'm totally fine with this.